src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -169,224 +176,668 @@ private sealed class ColInfoEx
+            }
+
+            private ValueGetter<VBuffer<Float>> GetTopic(IRow input, int iinfo)
+            {
+                var getSrc = RowCursorUtils.GetVecGetterAs<Double>(NumberType.R8, input, _srcCols[iinfo]);
+                var src = default(VBuffer<Double>);
+                var lda = _parent._ldas[iinfo];
+                int numBurninIter = lda.InfoEx.NumBurninIter;
+                bool reset = lda.InfoEx.ResetRandomGenerator;
+                return
+                    (ref VBuffer<Float> dst) =>
+                    {
+                        // REVIEW: This will work, but there are opportunities for caching
+                        // based on input.Counter that are probably worthwhile given how long inference takes.
+                        getSrc(ref src);
+                        lda.Output(in src, ref dst, numBurninIter, reset);
+                    };
+            }
+        }
+
+        public const string LoaderSignature = "LdaTransform";
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "LIGHTLDA",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(LdaTransform).Assembly.FullName);
+        }
+
+        private readonly TransformInfo[] _exes;
+        private readonly LdaState[] _ldas;
+        private readonly ColumnType[] _types;
+
+        private const string RegistrationName = "LightLda";
+        private const string WordTopicModelFilename = "word_topic_summary.txt";
+        internal const string Summary = "The LDA transform implements LightLDA, a state-of-the-art implementation of Latent Dirichlet Allocation.";
+        internal const string UserName = "Latent Dirichlet Allocation Transform";
+        internal const string ShortName = "LightLda";
+
+        private static (string input, string output)[] GetColumnPairs(ColumnInfo[] columns)
+        {
+            Contracts.CheckValue(columns, nameof(columns));
+            return columns.Select(x => (x.Input, x.Output)).ToArray();
+        }
+
+        internal LdaTransform(IHostEnvironment env, IDataView input, ColumnInfo[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(LdaTransform)), GetColumnPairs(columns))
+        {
+            _exes = new TransformInfo[columns.Length];
+            _types = new ColumnType[columns.Length];
+            _ldas = new LdaState[columns.Length];
+
+            for (int i = 0; i < columns.Length; i++)
+            {
+                var ex = new TransformInfo(Host, columns[i]);
+                _exes[i] = ex;
+                _types[i] = new VectorType(NumberType.Float, ex.NumTopic);
+            }
+            using (var ch = Host.Start("Train"))
+            {
+                Train(ch, input, _ldas);
```

Comment:
>Train [](start = 16, length = 5)

Pete suggested that we make training a private static method in the transform, and that we have a constructor for the transform that will accept the trained parameters as an input. 

This should make it easier to allow initialization of the transform using parameters coming from an external training routine. It is not something that we are exposing now, however.

This allows the transform to only keep the fields storing trained parameters.

Reply:
I have implemented something like this for whitening.

---
In reply to: [231639657](https://github.com/dotnet/machinelearning/pull/1410#discussion_r231639657) [](ancestors = 231639657)

Reply:
fixed. thanks for giving the context about this approach.

---
In reply to: [231640188](https://github.com/dotnet/machinelearning/pull/1410#discussion_r231640188) [](ancestors = 231640188,231639657)